### ISSUE TITLE
Party Presence is not consistent for the Leader

### ISSUE THREAD

**Issue by Meneleus** [heroiclabs/nakama#617]
## Description
When you create a party using `CreateParty` you receive a `Party` structure. You are listed as the `Leader` but you don't show up in the `Presences` field. You (the leader) *do* get a Join in the `ReceivedPartyPresence` event.

When you join a party using `JoinParty` you get a `ReceivedParty` event. Again, the `Leader` is specified correctly but again the `Leader` is not in the `Presences` field. You (the joiner) *do* get a Join in the `ReceivedPartyPresence` event, but the leader is *not* in that `Presences` field.

## Expected Result
The leader should be in `Presences`

## Context
- [X ] Unity

## Your Environment
- Nakama: 3.3.0

**Comment by lugehorsam** [comment]
I'm seeing the leader in the presences list after a second user joins, but not the joining user.

```
        [Fact]
        public async Task LeaderShouldBeInPresences()
        {
            var session1 = await _client.AuthenticateCustomAsync($"{Guid.NewGuid()}");
            var session2 = await _client.AuthenticateCustomAsync($"{Guid.NewGuid()}");

            var socket1 = Nakama.Socket.From(_client);
            var socket2 = Nakama.Socket.From(_client);

            await socket1.ConnectAsync(session1);
            await socket2.ConnectAsync(session2);

            var party = await socket1.CreatePartyAsync(true, 1);

            var joinedTcs = new TaskCompletionSource<IPartyPresenceEvent>();
            socket2.ReceivedPartyPresence += (evt) => {
                joinedTcs.SetResult(evt);
            };

            await socket2.JoinPartyAsync(party.Id);

            var presenceEvt = await joinedTcs.Task;

            Assert.Equal(2, presenceEvt.Joins.Count());
            Assert.True(presenceEvt.Joins.Any(join => join.UserId == session1.UserId));

            // give time for all presence events to fire
            await Task.Delay(1000);

            await socket1.CloseAsync();
            await socket2.CloseAsync();
        }
```

```
{"level":"info","ts":"2021-05-21T12:14:18.869Z","caller":"server/session_ws.go:81","msg":"New WebSocket session connected","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46","format":0}

{"level":"info","ts":"2021-05-21T12:14:18.878Z","caller":"server/session_ws.go:81","msg":"New WebSocket session connected","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46","format":0}

{"level":"debug","ts":"2021-05-21T12:14:18.889Z","caller":"server/pipeline.go:66","msg":"Received *rtapi.Envelope_PartyCreate message","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46","cid":"0","message":{"PartyCreate":{"open":true,"max_size":1}}}

{"level":"debug","ts":"2021-05-21T12:14:18.890Z","caller":"server/session_ws.go:391","msg":"Sending *rtapi.Envelope_Party message","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46","envelope":"cid:\"0\" party:{party_id:\"4f4de0ae-8721-4f57-9696-1de342a2d8de.nakama1\" open:true max_size:1 self:{user_id:\"88399b40-7cca-497b-943c-3d812c214aca\" session_id:\"116f363b-ba2e-11eb-ae3a-7106fdcb5b46\" username:\"fWiXTBSBtK\"} leader:{user_id:\"88399b40-7cca-497b-943c-3d812c214aca\" session_id:\"116f363b-ba2e-11eb-ae3a-7106fdcb5b46\" username:\"fWiXTBSBtK\"} presences:{user_id:\"88399b40-7cca-497b-943c-3d812c214aca\" session_id:\"116f363b-ba2e-11eb-ae3a-7106fdcb5b46\" username:\"fWiXTBSBtK\"}}"}

{"level":"debug","ts":"2021-05-21T12:14:18.890Z","caller":"server/tracker.go:834","msg":"Processing presence event","joins":1,"leaves":0}

{"level":"debug","ts":"2021-05-21T12:14:18.902Z","caller":"server/pipeline.go:66","msg":"Received *rtapi.Envelope_PartyJoin message","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46","cid":"1","message":{"PartyJoin":{"party_id":"4f4de0ae-8721-4f57-9696-1de342a2d8de.nakama1"}}}

{"level":"debug","ts":"2021-05-21T12:14:18.902Z","caller":"server/session_ws.go:391","msg":"Sending <nil> message","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46","envelope":"cid:\"1\""}

{"level":"debug","ts":"2021-05-21T12:14:18.902Z","caller":"server/tracker.go:834","msg":"Processing presence event","joins":1,"leaves":0}

{"level":"debug","ts":"2021-05-21T12:14:19.028Z","caller":"server/session_ws.go:200","msg":"Error reading message from client","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46","error":"websocket: close 1006 (abnormal closure): unexpected EOF"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:434","msg":"Cleaning up closed client connection","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:442","msg":"Cleaned up closed connection matchmaker","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"debug","ts":"2021-05-21T12:14:19.028Z","caller":"server/session_ws.go:200","msg":"Error reading message from client","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46","error":"websocket: close 1006 (abnormal closure): unexpected EOF"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:434","msg":"Cleaning up closed client connection","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:442","msg":"Cleaned up closed connection matchmaker","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:446","msg":"Cleaned up closed connection tracker","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:450","msg":"Cleaned up closed connection status registry","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:454","msg":"Cleaned up closed connection session registry","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:471","msg":"Closed client connection","uid":"80f583b0-ec3a-40cf-a189-e2454181cc7b","sid":"11708e57-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.029Z","caller":"server/session_ws.go:446","msg":"Cleaned up closed connection tracker","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.030Z","caller":"server/session_ws.go:450","msg":"Cleaned up closed connection status registry","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"info","ts":"2021-05-21T12:14:19.030Z","caller":"server/session_ws.go:454","msg":"Cleaned up closed connection session registry","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46"}

{"level":"debug","ts":"2021-05-21T12:14:19.030Z","caller":"server/tracker.go:834","msg":"Processing presence event","joins":0,"leaves":1}

{"level":"debug","ts":"2021-05-21T12:14:19.030Z","caller":"server/tracker.go:834","msg":"Processing presence event","joins":0,"leaves":1}

{"level":"info","ts":"2021-05-21T12:14:19.031Z","caller":"server/session_ws.go:471","msg":"Closed client connection","uid":"88399b40-7cca-497b-943c-3d812c214aca","sid":"116f363b-ba2e-11eb-ae3a-7106fdcb5b46"}
```

**Comment by lugehorsam** [comment]
@Meneleus I've confirmed this is a decoding issue on the dotnet client. Thank you.
